### PR TITLE
Ternary Expressions for Vue

### LINKED ISSUE #99: Vue generator breaks if schema's name has '-' in declaration
So, here how to reproduce the scenario, I am sharing the schema sample too. In the example below the event's name is ``input-box`` so when the generator runs it's breaking and when I remove the ``-`` from the name the component is generated successfully.

Possible Solutions
1. Configure the ``schema`` not to accept ``-`` in the name if it has an event. (which is a little tricky and I am not sure how we can achieve).
2. Change the logic that is being currently used for generating Event Handler names.

```
           {
                "type": "text",
                "name": "input-box",
                "events": {
                    "onClick": [
                        {
                            "type": "stateChange",
                            "modifies": "toggleComponent",
                            "newState": true
                        }
                    ]
                }
            }
```

~~~
{ SyntaxError: Unexpected token, expected "," (11:16)
   9 | 
  10 |   methods: {
> 11 |     handleInput-boxOnClick() {
     |                ^
  12 |       this.toggleComponent = true;
  13 |     }
  14 | 
    at t (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/parser-babylon.js:1:264)
    at Object.parse (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/parser-babylon.js:1:208428
)
    at Object.parse$2 [as parse] (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/standalone.js:12953:19)
    at coreFormat (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/standalone.js:16180:23)
    at format (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/standalone.js:16441:73)
    at formatWithCursor (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/standalone.js:16457:12)
    at /Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/standalone.js:30711:15
    at Object.format (/Users/a1290641/Documents/personal/teleport-code-generators/node_modules/prettier/standalone.js:30719:12)
    at Object.exports.generator [as js] (/Users/a1290641/Documents/personal/teleport-code-generators/src/core/builder/generators/js-ast-to-code.ts:16:21)
    at Builder.generateByType (/Users/a1290641/Documents/personal/teleport-code-generators/src/core/builder/index.ts:82:33)
  loc: { start: { line: 11, column: 16 } },
~~~

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
fixes #99 #93

### CODE REVIEW

**Comment by JayaKrishnaNamburu** [comment]
Hi @alexnm can you review the PR, sorry it became a little messy 😅

File: src/plugins/teleport-plugin-vue-base-component/utils.ts [hunk 1]
```diff
@@ -33,7 +34,7 @@ export const generateVueNodesTree = (
 
   if (events) {
     Object.keys(events).forEach((eventKey) => {
-      const methodName = `handle${capitalize(name)}${capitalize(eventKey)}`
+      const methodName = `handle${stringToCamelCase(name)}${stringToCamelCase(eventKey)}`
```
Comment (alexnm) [inline src/plugins/teleport-plugin-vue-base-component/utils.ts]: does this output: `handlebuttonclick` or `handleButtonClick`? the latter should be the case and I think you need to create another function that does `capitalize(stringToCamelCase(string))`. you can actually extract it in `string-utils` and name it `stringToUpperCamelCase`

File: src/plugins/teleport-plugin-vue-base-component/utils.ts [hunk 2]
```diff
@@ -70,23 +71,42 @@ export const generateVueNodesTree = (
...
-          htmlUtils.addChildNode(htmlNode, stateBranchNode)
-        })
+        const isBooleanState =
+          stateDefinitions[stateKey] && stateDefinitions[stateKey].type === 'boolean'
+        if (isBooleanState && stateBranches.length === 2) {
+          const conditionalStatement = createConditionalStatement(stateKey, stateBranches[0].value)
+          const consequentContent = stateBranches[0].content
+          const alternateContent = stateBranches[1].content
+          const consequentNode =
+            typeof consequentContent === 'string'
```
Comment (alexnm) [inline src/plugins/teleport-plugin-vue-base-component/utils.ts]: this specific ternary is repeated 3 times in this function, you could extract it into a "getNodeFromContent"

**Comment by JayaKrishnaNamburu** [comment]
hi @alexnm it's updated now

**Pull request merged**